en/22/11.md

```diff
@@ -0,0 +1,296 @@
+            &self, 
+            #[call_result] result: ManagedAsyncCallResult<Kitty>,
+            zombie_id: usize
+          ) {
+              match result {
+                  ManagedAsyncCallResult::Ok(kitty) => {},
+                  ManagedAsyncCallResult::Err(_) => {},
+              }
+          }
+
+          #[endpoint]
+          fn feed_on_kitty(
+            &self, 
+            zombie_id: usize,
+            kitty_id: usize,
+          ) {
+            let crypto_kitties_sc_address = self.crypto_kitties_sc_address().get();
+              self.kitty_proxy(crypto_kitties_sc_address)
+                  .get_kitty(kitty_id)
+                  .async_call()
+                  .with_callback(self.callbacks().get_kitty_callback(zombie_id))
+                  .call_and_exit();
+          }
+
+          #[proxy]
+          fn kitty_proxy(&self, to: ManagedAddress) -> crypto_kitties_proxy::Proxy<Self::Api>;
+      }
+---
+
+A part of our contract that used this proxy would look like this:
+
+```
+    #[endpoint]
+    fn call_adder_and_save_sum(&self) {
+      let caller = self.blockchain().get_caller();
+      let adder_contract_address = self.adder_contract_address().get();
+      self.adder_proxy(adder_contract_address)
+          .add(Biguint::from(2u32), Biguint::from(2u32))
+          .async_call()
+          .with_callback(self.callbacks().add_callback(&caller))
+          .call_and_exit();
+    }
+
+    #[callback]
+    fn add_callback(
+      &self, 
+      #[call_result] result: ManagedAsyncCallResult<BigUint>,
+      caller: &ManagedAddress
+    ) {
+        match result {
+            ManagedAsyncCallResult::Ok(sum) => {
+                self.sum(caller).set(sum);
+            },
+            ManagedAsyncCallResult::Err(_) => {
+                // this can only fail if the adder contract address is invalid
+                // nothing to revert in case of error
+            },
+        }
+    }
+
+    #[view(getSum)]
+    #[storage_mapper("sum")]
+    fn adder_contract_address(&self) -> SingleValueMapper<ManagedAddress>;
+
+    #[view(getSum)]
+    #[storage_mapper("sum")]
+    fn sum(&self, caller: &ManagedAddress) -> SingleValueMapper<BigUint>;
+
+```
+
+When calling another contract, if both contracts are in the same shard or not, we can make a direct call or a asynchronus call to one of their endpoints. In this context we can assume that the contracts are on diferent shards, reason why we used `async_call`.
+
+Since we are talking about asynchronus call we have no idea when we will recieve a response from the other contract (or if we will), reason why we will attach to this call another type of function defined by us, a callback which when will receive the result he will do something with it. The result is basically a fancy looking option with the result type of the called endpoint as a content or an error which in this case we will just ignore. A callback is just a basic function having the `#[callback]` annotation.
```

Comment:
```suggestion
Since we are talking about asynchronus call we have no idea when we will recieve a response from the other contract (or if we will), reason why we will attach to this call another type of function defined by us, a callback which when will receive the result it will do something with it. The result is basically a fancy looking option with the result type of the called endpoint as a content or an error which in this case we will just ignore. A callback is just a basic function having the `#[callback]` annotation.
```